### ISSUE TITLE
Type Mismatch for Conditions  in fsm.State

### ISSUE THREAD

**Issue by apjama** [viewflow/viewflow#468]
We noticed a possible issue with typing hint. 

Example to replicate

``` python
@state_machine.transition(
    source=States.PENDING,
    target=States.APPROVED,
    conditions=[is_approvable],
)
def approve(self, context):
    return TransitionResult(state=States.APPROVED)

def is_approvable(instance):
       return True
```

Mypy wil complain with:  `Expected type 'list[ThisObject | (object, object) -> bool] | None', got 'list[(instance: Any) -> bool]' instead ` 

Despite the type hint suggesting conditions take two arguments, they are actually called with only one argument.

`is_approvable` is a function that **can only** be called with one argument. 

Under the hood, this is how the conditions are called 

``` python
def conditions_met(self, instance: object) -> bool:
      """Checks if all conditions are met for this transition."""
      conditions = [
          (
              condition.resolve(instance.__class__)
              if isinstance(condition, ThisObject)
              else condition
          )
          for condition in self.conditions
      ]
      return all(map(lambda condition: condition(instance), conditions))
```

Is this intentional? Or have I just got the wrong end of the stick? 

If it's a genuine issue, I'm happy to put in a fix for it.

**Comment by kmmbvnr** [comment]
Yep, this looks like an issue

I've only started to add typing for viewflow definitions and would be glad to any help

**Comment by kmmbvnr** [comment]
Thank you, fixed in 6df5bed9ef539180ceeca6eed63b021d689b2ea would be released soon
